### PR TITLE
feature[ignore-typeKeys]: add possibility ignore typeKeys for timelin…

### PR DESCRIPTION
**Pull request by ArturRoze**
add possibility ignore typeKeys

### CODE REVIEW

File: xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java [hunk 1]
```diff
@@ -26,27 +31,31 @@ public class TimelineInterceptor extends HandlerInterceptorAdapter {
...
     private final List<String> ignoredHttpMethods;
+    private final List<String> ignoredTypeKeys;
 
     public TimelineInterceptor(
         TimelineEventProducer eventProducer,
         @Value("${application.tenant-ignored-path-list:true}") List<String> ignoredPatterns,
-        @Value("${application.timeline-ignored-http-methods:#{T(java.util.Collections).emptyList()}}") List<String> ignoredHttpMethods
-    ) {
+        @Value("${application.timeline-ignored-http-methods:#{T(java.util.Collections).emptyList()}}") List<String> ignoredHttpMethods,
+        @Value("${application.timeline-ignored-typekeys:#{T(java.util.Collections).emptyList()}}") List<String> ignoredTypeKeys
```
Comment (pkev777) [inline xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java:47]: For me better will be use kebab case as it was before: timeline-ignored-type-keys, what do you think?

File: xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java [hunk 2]
```diff
@@ -97,7 +106,38 @@ private boolean isIgnoredRequest(HttpServletRequest request) {
...
         }
-        return false;
+
+        String responseBody = getResponseContent(response);
+        String typeKey = getEntityField(responseBody, TYPE_KEY);
+
+        return isNotEmpty(ignoredTypeKeys) && ignoredTypeKeys.contains(typeKey);
+    }
+
+    private String getResponseContent(HttpServletResponse response) {
```
Comment (pkev777) [inline xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java:116]: This method is copy-paste and duplicated with TimelineEventProducer class. What do you think to extract this method to some utility class and make static?

File: xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java [hunk 3]
```diff
@@ -97,7 +106,38 @@ private boolean isIgnoredRequest(HttpServletRequest request) {
...
+        if (response instanceof HttpServletResponseWrapper
+            && ((HttpServletResponseWrapper) response).getResponse() instanceof ContentCachingResponseWrapper) {
+            return new String(((ContentCachingResponseWrapper) ((HttpServletResponseWrapper) response)
+                .getResponse()).getContentAsByteArray());
+        }
+        log.warn("Empty response content because of unsupported response class {}", response.getClass());
+        return "";
+    }
+
+    public String getEntityField(String entity, String field) {
```
Comment (pkev777) [inline xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java:129]: This method is copy-paste and duplicated with TimelineEventProducer class. What do you think to extract this method to some utility class and make static?

File: xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java [hunk 4]
```diff
@@ -97,7 +106,38 @@ private boolean isIgnoredRequest(HttpServletRequest request) {
                 }
             }
         }
-        return false;
+
+        String responseBody = getResponseContent(response);
```
Comment (pkev777) [inline xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/TimelineInterceptor.java:110]: Please add check: isNotEmpty(ignoredTypeKeys) before parsing http response. Motivation: if we don't have configured ignored types keys parsing response body operation was redundant

**Review by fpyshnyi — APPROVED** [review]

File: xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/util/JsonUtils.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+
+@Slf4j
+@UtilityClass
+public class JsonUtils {
+
+    public static String findField(String entity, String field, List<String> prefixes) {
```
[the remark this answers is not in the document]
Comment (ArturRoze) [reply xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/util/JsonUtils.java:14]: In our implementation, we are only interested in typeKey in the jason root.
Therefore, we do not search in the "data" and do not analyze the internal structure.
com.icthh.xm.commons.domainevent.utils.JsonUtil#extractIdAndTypeKey analyze all structure.

**Review by pkev777 — APPROVED** [review]

**Pull request merged**